src/app/app.component.ts

```diff
@@ -1,32 +1,41 @@
 import {ChangeDetectionStrategy, Component} from '@angular/core';
-import { RouterOutlet } from '@angular/router';
-import { ProductComponent} from "./product/product.component";
-import { CardComponent} from "./card/card.component";
+import {RouterOutlet} from '@angular/router';
+import {ProductComponent} from "./product/product.component";
+import {CardComponent} from "./card/card.component";
 import {NgForOf} from "@angular/common";
 import {products} from "./constants";
-import {Product} from "./interfaces";
+import {Card, Product} from "./interfaces";
 
 @Component({
-  selector: 'app-root',
-  templateUrl: './app.component.html',
-  styleUrl: './app.component.css',
-  changeDetection: ChangeDetectionStrategy.OnPush,
-  standalone: true,
-  imports: [RouterOutlet, ProductComponent, CardComponent, NgForOf],
+    selector: 'app-root',
+    templateUrl: './app.component.html',
+    styleUrl: './app.component.css',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    standalone: true,
+    imports: [RouterOutlet, ProductComponent, CardComponent, NgForOf],
 })
 export class AppComponent {
 
-  protected _products = products;
-  protected _cardList: Product[] = [];
+    protected _products = products;
+    protected _cardList: Card[] = [];
 
-  protected onProductSelected(product: Product): void {
-    if (product.quantity === 0) {
-      this._cardList.push(product);
+    protected onProductSelected(product: Product): void {
+        if (!this._cardList.find(element => element.id === product.id)) {
```

Comment:
Sépare cette fonction en 2

`const card = this._cardList.find(....);`

et selon le résultat,
- `addProductCard()`
- `incrementProductCard()`